Add new voices for Watson TTS

demikl opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
This PR adds new voices to be used from the IBM Watson TTS API.


## Type of change
Add voice names to the whitelist in the watson_tts component.

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
tts:
   - platform: watson_tts
     watson_apikey: !secret ibm_watson_tts_apikey
     voice: fr-FR_NicolasV3Voice

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- source for voice listing (needs an IBM Cloud account)

```sh
$ curl -X GET -u "apikey:XXXX" "https://api.eu-de.text-to-speech.watson.cloud.ibm.com/instances/YYYY/v1/voices" | jq '.voices[].name
```

- alternative link for listing the voices in a human readable format: https://cloud.ibm.com/docs/text-to-speech?topic=text-to-speech-voices#languageVoices

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository] : https://github.com/home-assistant/home-assistant.io/pull/17832

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

homeassistant commented:
Hello @demikl,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`6482dc278b9917950255b0a482371b08e6608159`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;core&#x2F;pull&#x2F;48722&#x2F;commits&#x2F;6482dc278b9917950255b0a482371b08e6608159) - This commit has something that looks like an email address (mlebaillif@oui.sncf). Maybe try linking that to GitHub?.
- [`2cf7dd2ca422a0116e7b66648b222a8b23b8cb9d`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;core&#x2F;pull&#x2F;48722&#x2F;commits&#x2F;2cf7dd2ca422a0116e7b66648b222a8b23b8cb9d) - This commit has something that looks like an email address (mlebaillif@oui.sncf). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

homeassistant commented:
Hi @demikl, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#48722).

Once you do that we will be able to review and accept this pull request.

Thanks!

rutkai commented:
Looks great, thanks for the list update!

demikl committed:
add voices as listed by:

$ curl -X GET -u "apikey:XXXX" "https://api.eu-de.text-to-speech.watson.cloud.ibm.com/instances/YYYY/v1/voices" | jq '.voices[].name'

homeassistant commented:
Hi @demikl, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#48722).

Once you do that we will be able to review and accept this pull request.

Thanks!

frenck commented on homeassistant/components/watson_tts/tts.py:
```diff
@@ -18,22 +18,32 @@
 # List from https://tinyurl.com/watson-tts-docs
 SUPPORTED_VOICES = [
     "ar-AR_OmarVoice",
+    "ar-MS_OmarVoice",
```
According to the docs, `ar-AR_OmarVoice` is deprecated by `AR_OmarVoice`.


According to the december update: <https://cloud.ibm.com/docs/text-to-speech?topic=text-to-speech-release-notes#December2020>

A lot of voices will be removed and replacements added. Currently, this PR only processes the new additions. Should it clean up as well?

demikl replied:
I preferred not to break existing user configuration, and scoping my changes to only adding new voices. Maybe another update, with deprecation warnings, could handle the removal of old voices ?

frenck replied:
IMHO this belongs in the same update. As additions made here, are not additions but renames.

demikl replied:
Sure, but according to the documentation you previously linked:

> It will continue to function for at least one year but might be removed at a future date.

So, in my opinion, it should be easier for users to be warned about a future deprecation of the voice they are using, instead of breaking their current configuration during next HA update.

frenck replied:
> it should be easier for users to be warned about a future deprecation of the voice they are using

Yes, thus a deprecation warning to the code should be added when those voices are used.

demikl replied:
I'll do it as advised. Thanks for your input.

demikl committed:
add warning for deprecated voices

demikl committed:
update default voice as previous one is now deprecated

frenck reviewed (approved):
Nice 👍 

Thanks, @demikl 🎉 


../Frenck 🚂

frenck commented on homeassistant/components/watson_tts/tts.py:
```diff
@@ -124,6 +161,12 @@ def __init__(self, service, supported_languages, default_voice, output_format):
 
+        if self.default_voice in DEPRECATED_VOICES:
+            _LOGGER.warning(
+                "Watson TTS voice %s is deprecated, it may be removed in the future.",
```
Small tweak...

```suggestion
                "Watson TTS voice %s is deprecated, it may be removed in the future",
```

frenck committed:
Log message without a full stop

frenck committed:
isort

MartinHjelmare commented on homeassistant/components/watson_tts/tts.py:
```diff
@@ -124,6 +162,12 @@ def __init__(self, service, supported_languages, default_voice, output_format):
         self.name = "Watson TTS"
 
+        if self.default_voice in DEPRECATED_VOICES:
+            _LOGGER.warning(
```
We try to avoid side effects in init methods. Please move the logging to `get_engine` instead.

frenck replied:
Right 👍 

Adjustment PR: #50716

MartinHjelmare reviewed (commented):
Please address the comment in a new PR. Thanks!
